src/neo/Consensus/ConsensusContext.cs

```diff
@@ -386,9 +386,21 @@ public void Reset(byte viewNumber)
                 CommitPayloads = new ConsensusPayload[Validators.Length];
                 if (LastSeenMessage == null)
                 {
-                    LastSeenMessage = new int[Validators.Length];
+                    LastSeenMessage = new Dictionary<ECPoint, int>();
                     for (int i = 0; i < Validators.Length; i++)
-                        LastSeenMessage[i] = -1;
+                        LastSeenMessage[Validators[i]] = (int)Snapshot.Height;
+                }
+                if (0 < Snapshot.Height && Blockchain.Singleton.GetBlock(Snapshot.Height - 1).NextConsensus != Blockchain.Singleton.GetBlock(Snapshot.Height).NextConsensus)
```

Comment:
Since most of the time `0 < Snapshot.Height` is true, perhaps this check could be the second.

Reply:
`Snapshot.Height` could be `0`.

Reply:
done.